src/box/iproto.cc

```diff
@@ -2382,6 +2424,35 @@ iproto_fill_stat(struct iproto_thread *iproto_thread,
 		mempool_count(&iproto_thread->iproto_msg_pool);
 }
 
+static inline void
+iproto_thread_detach_binaries(struct iproto_thread *iproto_thread)
+{
+	for (uint32_t i = 0; i < iproto_thread->binaries_count; i++)
+		evio_service_detach(&iproto_thread->binaries[i]);
+	iproto_threads->binaries_count = 0;
+}
+
+static inline void
+iproto_thread_check_listen_binaries(struct iproto_thread *iproto_thread)
+{
+	for (uint32_t i = 0; i < iproto_thread->binaries_count; i++) {
+		if (evio_service_is_active(&iproto_thread->binaries[i])) {
+			diag_set(ClientError, ER_UNSUPPORTED, "Iproto",
+				 "listen if service already active");
+			diag_raise();
```

Comment:
Nit: I'd make it return -1 so that raise occurs as further in call stack as it can be.

Reply:
I left it as it is, because if this function has a return code, then together with where it is called, you will have to do a line break, which looks terrible